### PR TITLE
Updated Acceptance Tests

### PR DESCRIPTION
**Pull request by clintonb**
- Logging in with LMS before all tests
- Skipping error page tests if the pages are not enabled

@benpatterson @muhammad-ammar @dsjen @rocha

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +27,7 @@ test_python: clean
 		$(PACKAGES)
 
 accept:
-	nosetests acceptance_tests --processes=2 --process-timeout=120 --exclude-dir=acceptance_tests/course_validation
+	nosetests -v acceptance_tests --processes=4 --process-timeout=120 --exclude-dir=acceptance_tests/course_validation
```
Comment (benpatterson) [inline Makefile]: Should these get dialed back down to 2 by default?

File: acceptance_tests/__init__.py [hunk 2]
```diff
@@ -1,205 +1,34 @@
...
+# LMS settings
+BASIC_AUTH_USERNAME = os.environ.get('BASIC_AUTH_USERNAME')
+BASIC_AUTH_PASSWORD = os.environ.get('BASIC_AUTH_PASSWORD')
+LMS_HOSTNAME = os.environ['LMS_HOSTNAME']
+LMS_USERNAME = os.environ['LMS_USERNAME']
+LMS_PASSWORD = os.environ['LMS_PASSWORD']
 
-class AnalyticsApiClientMixin(object):
-    api_client = None
+if ENABLE_OAUTH_TESTS and not(LMS_HOSTNAME and LMS_USERNAME and LMS_PASSWORD):
```
Comment (benpatterson) [inline acceptance_tests/__init__.py:31]: Thumbs up on this particular set of lines. Glad to see a check on the env vars!

File: acceptance_tests/mixins.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        Verify the user menu functions properly.
+        """
+        element = self.page.q(css='a.active-user.dropdown-toggle')
+        self.assertTrue(element.present)
+        self.assertEqual(element.attrs('aria-expanded')[0], 'false')
+
+        element.click()
+
+        # Check that the ARIA status was updated
+        self.assertEqual(element.attrs('aria-expanded')[0], 'true')
```
Comment (benpatterson) [inline acceptance_tests/mixins.py]: No need to change but a heads up that this pattern has resulted in some flaky tests on platform. You may end up needing a wait_for-style promise if this starts getting intermittent on you.

File: acceptance_tests/mixins.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    def assertSummaryPointValueEquals(self, data_selector, value):
+        """
+        Compares the value in the summary card the "value" argument.
+
+        Arguments:
+            data_selector (String): Attribute selector (ex. data-stat-type=current_enrollment)
+            tip_text (String): expected value
+        """
+        # Account for Django truncation
+        value = (value[:(MAX_SUMMARY_POINT_VALUE_LENGTH - 3)] + '...') if len(
```
Comment (benpatterson) [inline acceptance_tests/mixins.py]: nitpick: break into separate lines?

File: acceptance_tests/pages.py [hunk 5]
```diff
@@ -47,6 +48,26 @@ def is_browser_on_page(self):
...
+    def is_browser_on_page(self):
+        return self.browser.title.startswith('Log into')
+
+    def login(self, username, password):
+        self.q(css='input#email').fill(username)
+        self.q(css='input#password').fill(password)
+        self.q(css='button#submit').click()
+
+        # Wait for LMS to redirect to the dashboard
+        EmptyPromise(lambda: self.browser.title.startswith('Dashboard'), "LMS login redirected to dashboard").fulfill()
```
Comment (benpatterson) [inline acceptance_tests/pages.py]: nitpick: break into separate lines

**Comment by benpatterson** [comment]
Some comments but :+1: as-is.

File: Makefile (hunk 1 above)
Comment (rocha) [inline Makefile]: I keep thinking that deciding how many processes to run should be defined by the test runner, not the code. Is there something like a `.noserc` that we can set on Jenkins?

File: acceptance_tests/mixins.py (hunk 3 above)
Comment (rocha) [inline acceptance_tests/mixins.py]: I think this should be on the good practices. Given how tests run, we should about test that have race conditions.

File: acceptance_tests/mixins.py (hunk 4 above)
Comment (clintonb) [inline acceptance_tests/mixins.py]: Done.

File: acceptance_tests/mixins.py (hunk 3 above)
Comment (clintonb) [inline acceptance_tests/mixins.py]: Removed.

File: acceptance_tests/pages.py (hunk 5 above)
Comment (clintonb) [inline acceptance_tests/pages.py]: Extracted lambda to method.

File: acceptance_tests/mixins.py (hunk 3 above)
Comment (benpatterson) [inline acceptance_tests/mixins.py]: @rocha Yeah we have a start page for bok-choy but we have not yet produced a page on patterns and anti-patterns. We have a ticket for that but it's a bit further down the list at the moment.

**Pull request merged**